### ISSUE TITLE
Admin tool to show who's logged in now is needed

### ISSUE THREAD

**Issue by tmccormi** [LibreHealthIO/LibreEHR#343]
I wrote this SQL a while back - we could use a simple UI for admins that does this..

select user, IF( (mod(count(user),2)=1),'YES','NO') as logged_in from log where DATE_FORMAT(`date`,'%Y-%M-%d') = DATE_FORMAT(now(),'%Y-%M-%d') and (event = "login" or event = "logout") group by user

**Comment by aethelwulffe** [comment]
Very good idea, but the 7200 second (or more) activity sign-off is a 
factor in determining if someone is actually there or not.  I find that 
most users don't properly log off.  A two-hour delay is rough when you 
want to hut down mysql.

I suggest changing that default at least.  Another bit to consider might 
be an addition to the auto-logout that sends a message or some such to 
the user simply stating "you did not properly log out of your last 
session" to sort of nag them into hitting the log-out button in the future.

**Comment by tmccormi** [comment]
Not a bad idea, we should post that on the facebook page, I think....

PS: does autologout work in LibreEHR still?

**Comment by aethelwulffe** [comment]
Yes, I think autologout still functions...I seem to recall having logins go dead.  There is a way to test that.

**Comment by aethelwulffe** [comment]
Is it a valid approach to just check who has been active at all in...say...the last 30 minutes?  How about last activity for every active user instead?

**Comment by tmccormi** [comment]
I don't know if it makes much difference, but you would have to show the last activity, which could be "LOGOFF" in a best case

**Comment by aethelwulffe** [comment]
Starting here:
`function checklogin() {
    if (!$this->isLoggedIn())
        return false;
    else
        return true;
}


function isLoggedIn() {
    if (isset($_SESSION['user_id']) && isset($_SESSION['user_email'])) {

        if( !$this->isTimeOut() )
            return false;

        $where = array(
            'id' => $_SESSION['user_id'],
            'email' => $_SESSION['user_email']
        );

        //this is database class method
        $value['data'] = $dataBaseClass->login_access($where);

        if ($value['data'] != FALSE && $_SESSION['user_id'] == $value['data']->id && $_SESSION['user_email'] == $value['data']->email) {

            return true;
        }
        /*
          username or userlevel not exist or s_ecryption session invalid
          User not logged in.
         */
        else
            return false;
    }
    /**
     * cookie check for user login || you can escape this
     */
        else if (isset($_COOKIE['cookie_key']) && $_COOKIE['cookie_key'] != '') {

        $cookie_key = $_COOKIE['cookie_key'];
        $where = array(
            'cookie_key' => $cookie_key
        );
        $user_data = $databaseClass->login_access($where);

        if (!empty($user_data)) {
            $_SESSION['user_id'] = $user_data->id;
            $_SESSION['user_email'] = $user_data->email;

            return true;
        }
    } else {
        return false;
    }
}`
